File: TeeNew/Builder2009/Export_JPEG.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Export_JPEG.h"
#include "TeeJPEG.hpp"
#include "TeExport.hpp"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Base"
#pragma link "TeeJPEG"
#pragma resource "*.dfm"
TJPEGExportForm *JPEGExportForm;
//---------------------------------------------------------------------------
__fastcall TJPEGExportForm::TJPEGExportForm(TComponent* Owner)
        : TBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TJPEGExportForm::Button1Click(TObject *Sender)
{
  TeeSavePanel(__classid(TJPEGExportFormat),Chart1);
}
//---------------------------------------------------------------------------

void __fastcall TJPEGExportForm::Button2Click(TObject *Sender)
{
  TeeExport(this,Chart1);
}
//---------------------------------------------------------------------------

void __fastcall TJPEGExportForm::FormCreate(TObject *Sender)
{
  Series1->FillSampleValues(8);
}
//---------------------------------------------------------------------------

File: TeeNew/Export_JPEG.pas
unit Export_JPEG;
{$I TeeDefs.inc}

interface

uses
  {$IFNDEF LINUX}
  Windows, Messages,
  {$ENDIF}
  SysUtils, Classes,
  {$IFDEF CLX}
  QGraphics, QControls, QForms, QDialogs, QExtCtrls, QStdCtrls, QComCtrls,
  {$ELSE}
  Graphics, Controls, Forms, Dialogs, ExtCtrls, StdCtrls, ComCtrls,
  {$ENDIF}
  Base, TeEngine, Series, GanttCh, TeeProcs, Chart, ArrowCha;

type
  TJPEGExportForm = class(TBaseForm)
    Button1: TButton;
    Button2: TButton;
    Series1: TArrowSeries;
    procedure Button1Click(Sender: TObject);
    procedure Button2Click(Sender: TObject);
    procedure FormCreate(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

implementation

{$IFNDEF CLX}
{$R *.DFM}
{$ELSE}
{$R *.xfm}
{$ENDIF}

Uses TeeJPEG, TeExport;

procedure TJPEGExportForm.Button1Click(Sender: TObject);
begin
  TeeSavePanel(TJPEGExportFormat,Chart1);
end;

procedure TJPEGExportForm.Button2Click(Sender: TObject);
begin
  TeeExport(Self,Chart1);
end;

procedure TJPEGExportForm.FormCreate(Sender: TObject);
begin
  inherited;
  Series1.FillSampleValues(8);
end;

initialization
  RegisterClass(TJPEGExportForm);
end.
